src/reanimated2/Hooks.js

```diff
@@ -31,23 +31,22 @@ export function useSharedValue(init, shouldRebuild = true) {
   return ref.current.mutable;
 }
 
-export function useMapper(fun, inputs = [], outputs = [], dependencies = []) {
-  useEffect(() => {
-    const mapperId = startMapper(fun, inputs, outputs);
-    return () => {
-      stopMapper(mapperId);
-    };
-  }, dependencies);
-}
-
 export function useEvent(handler, eventNames = [], rebuild = false) {
   const initRef = useRef(null);
   if (initRef.current === null) {
     initRef.current = new WorkletEventHandler(handler, eventNames);
   } else if (rebuild) {
     initRef.current.updateWorklet(handler);
   }
-  return initRef.current;
+
+  useEffect(() => {
+    return () => {
+      initRef.current.unregisterFromEvents();
```

Comment:
`unregisterFromEvents` is already called in `createAimatedComponent` – you should've explained why is it needed to be added here. It seem more appropriate to control registration/unreg in one place instead of having that logic and responsibility split across different components. Currently `createAnimatedComponnt` is responsible for registering for the events and so it should also be responsible for the cleanup.